lib/matplotlib/widgets.py

```diff
@@ -613,43 +607,94 @@ def _clicked(self, event):
                 self.set_active(i)
                 break
 
-    def set_active(self, index):
+    def set_active(self, index, state=_TOGGLE):
         """
         Directly (de)activate a check button by index.
 
-        *index* is an index into the original label list
-            that this object was constructed with.
-            Raises ValueError if *index* is invalid.
+        Default behaviour is to toggle the state, which can be controlled with
+        state=bool flag.
 
         Callbacks will be triggered if :attr:`eventson` is True.
 
+        Parameters
+        ----------
+        index : int
+            index into the original label list
+
+        state : bool
```

Comment:
Suggestion:
~~~
       state : bool, optional
            Determines the target state of the button. Boolean values
            set the state explicitly. If not given, the state gets toggled.